drivers/afe/ad4110/ad4110.c

```diff
@@ -712,16 +704,33 @@ int32_t ad4110_setup(struct ad4110_dev **device,
 	/* SPI */
 	ret = spi_init(&dev->spi_dev, &init_param.spi_init);
 
+	dev->afe_crc_en = AD4110_AFE_CRC_DISABLE;
+	dev->adc_crc_en = AD4110_ADC_CRC_DISABLE;
 	dev->addr = init_param.addr;
-	dev->adc_crc_en = init_param.adc_crc_en;
-	dev->afe_crc_en = init_param.afe_crc_en;
 	dev->nready_pin = init_param.nready_pin;
 	dev->irq_desc = init_param.irq_desc;
 
 	/* Device Settings */
 	ad4110_spi_do_soft_reset(dev);
 	mdelay(10);
 
+	if(init_param.afe_crc_en != AD4110_AFE_CRC_DISABLE) {
+		ret |= ad4110_spi_int_reg_write(dev,
+						A4110_AFE,
+						AD4110_REG_AFE_CNTRL1,
+						AD4110_REG_AFE_CNTRL1_CRC_EN);
+	}
+	dev->afe_crc_en = init_param.afe_crc_en;
+
+	if(init_param.adc_crc_en != AD4110_ADC_CRC_DISABLE) {
+		ret |= ad4110_spi_int_reg_write_msk(dev,
+						    A4110_ADC,
+						    AD4110_REG_ADC_INTERFACE,
+						    AD4110_ADC_CRC_EN(init_param.adc_crc_en),
```

Comment:
initialization parameters are passed to the device descriptor usually at the start of the init function. Afterwards you operate only with the device descriptor.